### PR TITLE
Displaydashboard

### PR DESCRIPTION
**Pull request by sam-ao**
This change is [Reviewable](https://reviewable.io/reviews/ucsolarcarteam/epsilon-dashboard/3)

### CODE REVIEW

File: ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    QLabel& busVoltageLabel();
+    QLabel& busCurrentLabel();
+
+    QLabel& powerInLabel();
+    QLabel& powerOutLabel();
+    QLabel& netPowerLabel();
+
+private:
+    Ui::DisplayDashboardUI *ui;
```
Comment (bill-luu) [inline ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h]: Underscore after private members, so `ui_`

File: ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+{
+    delete ui;
+}
+
+QLabel& DisplayDashboardUI::actualSpeedLabel()
+{
+    return *ui->actualSpeedLabel;
+}
+
+
```
Comment (bill-luu) [inline ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp]: You have one too many new lines here

File: ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+{
+    return *ui->setCurrentLabel;
+}
+
+QLabel& DisplayDashboardUI::motorPowerLabel()
+{
+    return *ui->motorPowerLabel;
+}
+
+
```
Comment (bill-luu) [inline ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp]: Too many new lines here too

File: Gen5Dashboard.pro [hunk 4]
```diff
@@ -150,12 +150,13 @@ HEADERS  += \
...
-    ViewLayer/CompactUI/CompactUI.h \
-    ViewLayer/CompactView/CompactView.h \
-    ViewLayer/CompactUI/I_CompactUI.h
+    ViewLayer/DisplayDashboardView/DisplayDashboardView.h \
+    ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h \
+    ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h
 
 FORMS    += \
-    CompactUI.ui
+    CompactUI.ui \
```
Comment (bill-luu) [inline Gen5Dashboard.pro]: I don't think we'll be using the CompactUI anymore, so we might as well get rid of this too.

**Comment by enochtsang** [comment]
Reviewed 47 of 53 files at r1, 5 of 7 files at r2.
Review status: 51 of 52 files reviewed at latest revision, 12 unresolved discussions.

---

*[Gen5Dashboard.pro.user, line 1 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2zQDP7HGsbsTezhrt:-KS2zQDP7HGsbsTezhru:bowh0rk) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Gen5Dashboard.pro.user#L1)):*
> ```
> <?xml version="1.0" encoding="UTF-8"?>
> ```

This file needs to be ignored, please add *.pro.* to the `.gitignore.`

---

*[Makefile, line 0 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS36GyhdvKxbM037GRH:-KS36Gyi3hNFBjtW0oQ4:bekq0ow) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Makefile#L0)):*
The Makefile should be ignored and removed from the repo.

---

*[ViewLayer/ViewContainer.cpp, line 35 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37OI2MJ360soh-OLp:-KS37OI2MJ360soh-OLq:b-37ks2d) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/ViewContainer.cpp#L35)):*
> ```cpp
> ViewContainer::ViewContainer(PresenterContainer& presenterContainer)
> : DisplayDashboardUI_(new DisplayDashboardUI())
> , DisplayDashboardView_(new DisplayDashboardView(
> ```

How come we have `DisplayView` and `DisplayDashboardView` I feel like we hsoudl only need one of those, prefereably `DisplayDashboardView`.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h, line 9 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37l2gXevETYUpEEZX:-KS37l2hE2_tQbfO9rWQ:butgsbx) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h#L9)):*
> ```objc
>     class DisplayDashboardUI;
> }
> class DisplayDashboardUI : public I_SolarCarWindow, public I_DisplayDashboardUI
> ```

I_SolarCarWindow is legacy code, this is not necessary let's delete it.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 4 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2xuXkYOH6C5AO798y:-KS2xuXlalfal1DIliQy:b-ciom1o) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L4)):*
> ```objc
> 
> #include <QLabel>
> #include <QTextEdit>
> ```

Unused?

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 7 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2yJWAYsdBIAuqdhF2:-KS2yJWAYsdBIAuqdhF3:b-wmzuww) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L7)):*
> ```objc
> #include <QProgressBar>
> 
> class I_DisplayDashboardUI
> ```

The interface files should be inheriting QWidget. This is to avoid multiple inheritance, a linear inhertiance tree is easier to think about. [Example](https://github.com/UCSolarCarTeam/Delta-Desktop-Client-Telemetry/blob/master/ViewLayer/PlaybackUI/I_PlaybackUI.h)

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 14 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2xRRd7vWF_KKW4LZZ:-KS2xRRd7vWF_KKW4LZ_:b-d9alit) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L14)):*
> ```objc
>     virtual QLabel& actualSpeedLabel() = 0;
> 
>     virtual QLabel& arrayOneVoltageLabel() = 0;
> ```

Gen 5 counting was confusing sometimes because we counted starting from 0 sometimes, and 1 sometimes. So change this to start from 0 please. Also I think using the actual digit will look nicer since less code.

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp, line 9 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2ymXypxo4X0zEUAn4:-KS2ymXypxo4X0zEUAn5:bukfqut) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp#L9)):*
> ```cpp
> #include "PresenterLayer/VehiclePresenter/VehiclePresenter.h"
> #include "ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h"
> #include <QDateTime>
> ```

Necessary?

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.h, line 41 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37vPnrisa11_yZS_8:-KS37vPnrisa11_yZS_9:b-bvtxu9) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardView/DisplayDashboardView.h#L41)):*
> ```objc
>     I_DisplayDashboardUI& ui_;
> 
> private slots:
> ```

no need for this line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by enochtsang** [comment]
Review status: 51 of 52 files reviewed at latest revision, 13 unresolved discussions.

---

*[Resources/Thermometer.bmp, line 0 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS3945IpfQN0vyDVQpz:-KS3945IpfQN0vyDVQq-:b-f3ntb1) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Resources/Thermometer.bmp#L0)):*
Is this file necessary?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by bill-luu** [comment]
Review status: 51 of 52 files reviewed at latest revision, 11 unresolved discussions.

---

*[Gen5Dashboard.pro, line 158 at r1](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2wVdyRSGRrVCdaTzm-r1-158:-KS8IiLTPWiFXJfa2dFd:bascfpb) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/f54c747271d1dcdb68f8202880aa2e583661fea7/Gen5Dashboard.pro#L158)):*
<details><summary><i>Previously, bill-luu wrote…</i></summary>
> I don't think we'll be using the CompactUI anymore, so we might as well get rid of this too.
</details>
The actual `.ui` file should be deleted too if you aren't using it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by enochtsang** [comment]
Reviewed 1 of 7 files at r2, 23 of 23 files at r3.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[uiresources.qrc, line 39 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScNlrp5uR0lwpwbwX2:-KScNlrqlMCDdOHknGr3:b-garvdx) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/uiresources.qrc#L39)):*
> ```
>         <file>Resources/HighHeadlightIndicator.png</file>
>         <file>Resources/ArrayFourBackground.png</file>
>         <file>Resources/ArrayOneBackground.png</file>
> ```

Let's change these to start counting from 0.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 22 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScLW3sHyrKK7hHlLgH:-KScLW3sHyrKK7hHlLgI:b-dk8kxo) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L22)):*
> ```objc
>     virtual QLabel& array1CurrentLabel() = 0;
>     virtual QLabel& array2CurrentLabel() = 0;
>     virtual QLabel& array3CurrentLabel() = 0;
> ```

There were changes, there's only gonna be 3 mppts now.

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp, line 25 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScLcjvPFYKZmUWxKn1:-KScLcjwfVuR3d3mckn4:b-j3gzmp) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp#L25)):*
> ```cpp
>    , ui_(ui)
> {
>     ui_.show();
> ```

This should be show maximized. For the pi, it needs to be not full screen, and also be borderless. look up window hints to do this.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by bill-luu** [comment]
Reviewed 37 of 53 files at r1, 22 of 23 files at r3.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by sam-ao** [comment]
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[uiresources.qrc, line 39 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScNlrp5uR0lwpwbwX2:-KSrDvver-rhUvAKiAQL:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/uiresources.qrc#L39)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> Let's change these to start counting from 0.
</details>
Done.

---

*[ViewLayer/ViewContainer.cpp, line 35 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37OI2MJ360soh-OLp:-KSrCBR5HP31Y5BuElOl:bs7wumb) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/ViewContainer.cpp#L35)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> How come we have `DisplayView` and `DisplayDashboardView` I feel like we hsoudl only need one of those, prefereably `DisplayDashboardView`.
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp, line 20 at r1](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2wVdz6l_mHn_ZXkGg-r1-20:-KSrCQ5abIdv3Psw9rw2:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/f54c747271d1dcdb68f8202880aa2e583661fea7/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp#L20)):*
<details><summary><i>Previously, bill-luu wrote…</i></summary>
> Let's just have one new line between the functions here
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h, line 62 at r1](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2wVdz6l_mHn_ZXkGh-r1-62:-KSrCR12nek0Qh6bKedi:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/f54c747271d1dcdb68f8202880aa2e583661fea7/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h#L62)):*
<details><summary><i>Previously, bill-luu wrote…</i></summary>
> Underscore after private members, so `ui_`
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 22 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScLW3sHyrKK7hHlLgH:-KSrDsRV-3Y3USdkTNzn:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L22)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> There were changes, there's only gonna be 3 mppts now.
</details>
Done.

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.h, line 41 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37vPnrisa11_yZS_8:-KSrDrGl41W73255pkR1:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardView/DisplayDashboardView.h#L41)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> no need for this line
</details>
Done.

---

*[Resources/Thermometer.bmp, line  at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS3945IpfQN0vyDVQpz:-KSrDqq9-DVynirT_I_o:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Resources/Thermometer.bmp#L)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> Is this file necessary?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by sam-ao** [comment]
Review status: 53 of 62 files reviewed at latest revision, 12 unresolved discussions.

---

*[Gen5Dashboard.pro, line 158 at r1](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2wVdyRSGRrVCdaTzm-r1-158:-KSrE51EP9k8EwHd-V68:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/f54c747271d1dcdb68f8202880aa2e583661fea7/Gen5Dashboard.pro#L158)):*
<details><summary><i>Previously, bill-luu wrote…</i></summary>
> The actual `.ui` file should be deleted too if you aren't using it.
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h, line 9 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS37l2gXevETYUpEEZX:-KSrF0aJA8ZHT3fZNYr3:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.h#L9)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> I_SolarCarWindow is legacy code, this is not necessary let's delete it.
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 4 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2xuXkYOH6C5AO798y:-KSrF22bg2ySruziFw5x:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L4)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> Unused?
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 7 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2yJWAYsdBIAuqdhF2:-KSrF320M7MlXYyGclQg:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L7)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> The interface files should be inheriting QWidget. This is to avoid multiple inheritance, a linear inhertiance tree is easier to think about. [Example](https://github.com/UCSolarCarTeam/Delta-Desktop-Client-Telemetry/blob/master/ViewLayer/PlaybackUI/I_PlaybackUI.h)
</details>
Done.

---

*[ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h, line 14 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2xRRd7vWF_KKW4LZZ:-KSrF1hAOCk9RvaRv04-:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardUI/I_DisplayDashboardUI.h#L14)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> Gen 5 counting was confusing sometimes because we counted starting from 0 sometimes, and 1 sometimes. So change this to start from 0 please. Also I think using the actual digit will look nicer since less code.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by sam-ao** [comment]
Review status: 52 of 62 files reviewed at latest revision, 9 unresolved discussions.

---

*[Makefile, line  at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS36GyhdvKxbM037GRH:-KSrHu7dXmxP2kDPl9X-:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Makefile#L)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> The Makefile should be ignored and removed from the repo.
</details>
Done.

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp, line 9 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2ymXypxo4X0zEUAn4:-KSrF92HquUeNSUnmIMO:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp#L9)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> Necessary?
</details>
Done.

---

*[Gen5Dashboard.pro.user, line 1 at r2](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KS2zQDP7HGsbsTezhrt:-KSrHwftt1s_MyeEAW95:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/de643ae1ae9bf99b14d08a906b6a2c04a6425a37/Gen5Dashboard.pro.user#L1)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> This file needs to be ignored, please add *.pro.* to the `.gitignore.`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by sam-ao** [comment]
Review status: 52 of 62 files reviewed at latest revision, 8 unresolved discussions.

---

*[ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp, line 25 at r3](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KScLcjvPFYKZmUWxKn1:-KSrONFmqHvk0WnufW4i:b-896fix) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/e0a616fe83ab586165045cdad63417b0f3c37776/ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp#L25)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> This should be show maximized. For the pi, it needs to be not full screen, and also be borderless. look up window hints to do this.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by enochtsang** [comment]
Reviewed 1 of 7 files at r2, 13 of 14 files at r4.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp, line 8 at r4](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KSrb46sZTEBNJHSmtOj:-KSrb46sZTEBNJHSmtOk:b-chbuog) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/0014c2a35e3af218bc7d6616a22a29c0980c6887/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp#L8)):*
> ```cpp
> {
>     ui_->setupUi(this);
>     QWidget::setWindowFlags(Qt::FramelessWindowHint);
> ```

I recall there being two flags that needs to be set, when this compiles is there nothing around it? No border or title?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Comment by sam-ao** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp, line 8 at r4](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3#-KSrb46sZTEBNJHSmtOj:-KSrcbvb9qXtyp7qF5-h:b-5doidw) ([raw file](https://github.com/ucsolarcarteam/epsilon-dashboard/blob/0014c2a35e3af218bc7d6616a22a29c0980c6887/ViewLayer/DisplayDashboardUI/DisplayDashboardUI.cpp#L8)):*
<details><summary><i>Previously, enochtsang (Enoch Tsang) wrote…</i></summary>
> I recall there being two flags that needs to be set, when this compiles is there nothing around it? No border or title?
</details>


---


*Comments from [Reviewable](https://reviewable.io:443/reviews/ucsolarcarteam/epsilon-dashboard/3)*

**Pull request merged**